### PR TITLE
DEV: Convert composer from named outlet to service/component

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Named outlets are deprecated and will be removed in Ember 4.x.

(to be rebased-and-merge so that the file renames remain in a separate commit)

### CODE REVIEW

File: app/assets/javascripts/discourse/app/routes/discourse.js [hunk 1]
```diff
@@ -53,7 +54,7 @@ const DiscourseRoute = Route.extend({
   },
 
   openTopicDraft() {
-    const composer = this.controllerFor("composer");
+    const composer = getOwner(this).lookup("service:composer");
```
[the remark this answers is not in the document]
Comment (CvX) [reply app/assets/javascripts/discourse/app/routes/discourse.js:57]: Ah, got it! 😃 Are those in components (popup-input-tip, sidebar/user/community-section, user-stream) safe to be converted into injections?

File: app/assets/javascripts/discourse/app/services/composer.js [hunk 2]
```diff
@@ -0,0 +1,1647 @@
...
+    return promise;
+  }
+
+  @action
+  postWasEnqueued(details) {
+    showModal("post-enqueued", {
+      model: details,
+      title: "review.approval.title",
+    });
+  }
```
[the remark this answers is not in the document]
Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/services/composer.js:1163]: There are a couple other changes - I'd recommend reviewing [this commit](https://github.com/discourse/discourse/pull/20720/commits/14239354d1b2ca5b391b3acde5c68237b9838ae8#diff-195d780addd0c3e7e5cf98f0b662aa509197ecc7a57e522d244832017f323f99) (I separated out the rename to make git history useful)

File: app/assets/javascripts/discourse/app/routes/discourse.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/routes/discourse.js:57]: Good call - I updated those components to use injections 👌

**Review by CvX — APPROVED** [review]
🚀

**Pull request merged**